tests/components/mikrotik/test_hub.py

```diff
@@ -0,0 +1,179 @@
+"""Test Mikrotik hub."""
+from asynctest import patch
+import librouteros
+
+from homeassistant import config_entries
+from homeassistant.components import mikrotik
+
+from . import ARP_DATA, DHCP_DATA, MOCK_DATA, MOCK_OPTIONS, WIRELESS_DATA
+
+from tests.common import MockConfigEntry
+
+
+async def setup_mikrotik_entry(hass, **kwargs):
+    """Set up Mikrotik intergation successfully."""
+    support_wireless = kwargs.get("support_wireless", True)
+    dhcp_data = kwargs.get("dhcp_data", DHCP_DATA)
+    wireless_data = kwargs.get("wireless_data", WIRELESS_DATA)
+
+    def mock_command(self, cmd, params=None):
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.IS_WIRELESS]:
+            return support_wireless
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.DHCP]:
+            return dhcp_data
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.WIRELESS]:
+            return wireless_data
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.ARP]:
+            return ARP_DATA
+        return {}
+
+    config_entry = MockConfigEntry(
+        domain=mikrotik.DOMAIN, data=MOCK_DATA, options=MOCK_OPTIONS
+    )
+    config_entry.add_to_hass(hass)
+
+    if "force_dhcp" in kwargs:
+        config_entry.options["force_dhcp"] = True
+
+    if "arp_ping" in kwargs:
+        config_entry.options["arp_ping"] = True
+
+    with patch("librouteros.connect"), patch.object(
+        mikrotik.hub.MikrotikData, "command", new=mock_command
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        return hass.data[mikrotik.DOMAIN][config_entry.entry_id]
+
+
+async def test_hub_setup_successful(hass):
+    """Successful setup of Mikrotik hub."""
+    with patch(
+        "homeassistant.config_entries.ConfigEntries.async_forward_entry_setup",
+        return_value=True,
+    ) as forward_entry_setup:
+        hub = await setup_mikrotik_entry(hass)
+
+    assert hub.config_entry.data == {
+        mikrotik.CONF_NAME: "Mikrotik",
+        mikrotik.CONF_HOST: "0.0.0.0",
+        mikrotik.CONF_USERNAME: "user",
+        mikrotik.CONF_PASSWORD: "pass",
+        mikrotik.CONF_PORT: 8278,
+        mikrotik.CONF_VERIFY_SSL: False,
+    }
+    assert hub.config_entry.options == {
+        mikrotik.hub.CONF_FORCE_DHCP: False,
+        mikrotik.CONF_ARP_PING: False,
+        mikrotik.CONF_DETECTION_TIME: 300,
+    }
+
+    assert hub.api.available is True
+    assert hub.signal_update == "mikrotik-update-0.0.0.0"
+    assert forward_entry_setup.mock_calls[0][1] == (hub.config_entry, "device_tracker")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4918093)